src/server/services/govuk-notify.ts

```diff
@@ -186,23 +186,39 @@ export async function sendAnnualReviewCompletedEmail(
   }
 }
 
-export async function sendEmails<P extends { [key: string]: any }>(
+export async function sendEmails<Personalisation extends { [key: string]: any }>(
   templateId: string,
   emailAddresses: string[],
-  options: SendEmailOptions<P>
+  options: SendEmailOptions<Personalisation>,
+  logLabel: string = ""
 ) {
   const notifyClient = getNotifyClient();
 
   logger.info(
-    `Template ID: ${templateId}, to emails ${emailAddresses}, with sendEmailOption ${JSON.stringify(options)}`,
+    `${logLabel} Template ID: ${templateId}, to emails ${emailAddresses}, with sendEmailOption ${JSON.stringify(
+      options
+    )}`,
     { method: "sendEmails" }
   );
 
   const requests = emailAddresses.map(async (emailAddress) => {
     return await notifyClient.sendEmail(templateId, emailAddress, options);
   });
 
-  return await Promise.allSettled(requests);
+  const settled = await Promise.allSettled(requests);
+
+  settled.filter(hasNotifyError).forEach((reject) => {
+    // @ts-ignore
+    logger.error(`${logLabel} Template ID: ${templateId} rejected with from notify API ${reject.reason}`, {
+      method: "sendEmails",
+    });
+  });
+
+  return await Promise.any(requests);
```

Comment:
Out of curiosity what is the benefit of returning a promise here?

Reply:
So the calling function can decide what it wants to do with the resolved value (or the error). 

You might want to use the resolved value for adding audit logs - we don't do this but notify returns a bunch of stuff like the ID, so you can use it to do a lookup later.  

And generally, for testability. Returning a value is more helpful than void functions. Void functions are more difficult to test, to test a void function you may have to on calls made within the function, but if the function returns you can just test the return value is what you expect.